zcash_client_sqlite/src/lib.rs

```diff
@@ -245,6 +245,32 @@ impl<C: Borrow<rusqlite::Connection>, P: consensus::Parameters> WalletRead for W
     type Error = SqliteClientError;
     type AccountId = AccountId;
 
+    fn validate_seed(
+        &self,
+        account_id: Self::AccountId,
+        seed: &SecretVec<u8>,
+    ) -> Result<bool, Self::Error> {
+        self.get_unified_full_viewing_keys().and_then(|keys| {
+            keys.get(&account_id).map_or(Ok(false), |ufvk| {
+                let usk = UnifiedSpendingKey::from_seed(
+                    &self.params,
+                    &seed.expose_secret()[..],
```

Comment:
Should this be passed further down the call chain as a `&SecretVec<u8>`?

Reply:
If we want to do so, it should be in a separate PR.

Reply:
All this practically does is force the caller to use `SecretVec` for its own memory management; given that we do not copy or persist the seed inside `UnifiedSpendingKey::from_seed`, and don't perform any comparisons or branching on it, passing the data around as a slice is fine.